### PR TITLE
[DATA-1345]: Use latest openapi generator, create part only if it doesn't already exist

### PR DESCRIPTION
**Pull request by therockstorm**
## Possible Regressions

n/a

## Dependencies

n/a

### CODE REVIEW

File: client/vertex-client.ts [hunk 1]
```diff
@@ -69,14 +61,20 @@ export class VertexClient {
...
 
-  private constructor({ auth, baseOptions, basePath, config }: CtorArgs) {
+  private constructor({ auth, baseOptions, basePath, token }: CtorArgs) {
     this.auth = auth;
-    this.baseOptions = baseOptions || {};
-    this.basePath = basePath;
+    this.token = token;
+    this.tokenFetchedEpochMs = nowEpochMs();
+    const config = new Configuration({
+      accessToken: this.accessTokenRefresher,
```
Comment (therockstorm) [inline client/vertex-client.ts:73]: The latest generator version allows us to pass a function that returns a promise 😎

File: client/helpers/parts.ts [hunk 2]
```diff
@@ -24,6 +25,38 @@ export const createPartFromFile = async (
...
 
   const req = args.createPartReq(fileId);
+  const suppliedId = req.data.attributes.suppliedId;
+  const suppliedRevisionId = req.data.attributes.suppliedRevisionId;
+
+  // TODO: Update once we can filter by both part and part-revision suppliedIds with one API call
+  const getPartRes = await args.client.parts.getParts(undefined, 1, [
+    suppliedId,
+  ]);
+  throwOnError(getPartRes, `Error getting parts by suppliedId '${suppliedId}'`);
```
Comment (danschultz) [inline client/helpers/parts.ts:35]: Is it possible to wrap the Axios instance that is used by the client to provide this behavior? Seems like this pattern of throwing on non-success codes will be difficult to enforce moving forward.

Comment (therockstorm) [reply client/helpers/parts.ts:35]: We can, but that would catch the errors for anyone using the client and I'm not sure everyone would want to handle the errors in the same way.

Comment (danschultz) [reply client/helpers/parts.ts:35]: Maybe food for thought then, but not throwing an error is almost never what I want. I understand why these low-level request libraries don't throw errors. As a consumer though, this default approach basically forces me to check the response code for every API call I make, then switch on the result in order to parse the success or error data.

IMO, `Promise.then().catch()` is a better pattern for this. Could also give the consumer an option on what style they want.

Comment (therockstorm) [reply client/helpers/parts.ts:35]: I added a note for myself to give this some thought and can follow up with you.

**Review by danschultz — APPROVED** [review]

File: client/utils.ts [hunk 3]
```diff
@@ -67,6 +83,9 @@ export const pollQueuedJob = async <T extends { data: { id: string } }>(
 export const prettyJson = (obj: unknown): string =>
   JSON.stringify(obj, null, 2);
 
+export const sleep = async (ms: number) =>
+  new Promise((resolve) => setTimeout(resolve, ms));
```
[the remark this answers is not in the document]
Comment (therockstorm) [reply client/utils.ts]: Updated

**Review by danschultz — APPROVED** [review]

**Pull request merged**